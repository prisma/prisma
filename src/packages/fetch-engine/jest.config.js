module.exports = {
  preset: 'ts-jest',
  testEnvironment: 'node',
  collectCoverage: process.env.CI ? true : false,
  coverageReporters: ['clover'],
  coverageDirectory: 'src/__tests__/coverage',
  testMatch: ['**/src/__tests__/**/*.test.ts'],
  collectCoverageFrom: ['src/**/*.ts', '!**/__tests__/**/*'],
  globals: {
    'ts-jest': {
      packageJson: 'package.json',
    },
  },
}
