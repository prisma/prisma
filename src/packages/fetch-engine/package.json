{
  "name": "@prisma/fetch-engine",
  "version": "2.11.0-dev.20",
  "main": "dist/index.js",
  "types": "dist/index.d.ts",
  "license": "Apache-2.0",
  "author": "Tim Suchanek <suchanek@prisma.io>",
  "devDependencies": {
    "@prisma/engines-version": "2.12.0-18.cf0680a1bfe8d5e743dc659cc7f08009f9587d58",
    "@types/fs-extra": "9.0.4",
    "@types/jest": "26.0.15",
    "@types/node": "12.19.6",
    "@types/node-fetch": "2.5.7",
    "@types/progress": "2.0.3",
    "@typescript-eslint/eslint-plugin": "4.8.1",
    "@typescript-eslint/parser": "4.8.1",
    "del": "6.0.0",
    "eslint": "7.14.0",
    "eslint-config-prettier": "6.15.0",
    "eslint-plugin-eslint-comments": "3.2.0",
    "eslint-plugin-jest": "24.1.3",
    "eslint-plugin-prettier": "3.1.4",
    "husky": "4.3.0",
    "jest": "26.6.3",
    "lint-staged": "10.5.1",
    "prettier": "2.2.0",
    "ts-jest": "26.4.4",
    "strip-ansi": "6.0.0",
    "typescript": "4.1.2"
  },
  "dependencies": {
    "@prisma/debug": "workspace:*",
    "@prisma/get-platform": "workspace:*",
    "chalk": "^4.0.0",
    "execa": "^4.0.0",
    "find-cache-dir": "^3.3.1",
    "hasha": "^5.2.0",
    "http-proxy-agent": "^4.0.1",
    "https-proxy-agent": "^5.0.0",
    "make-dir": "^3.0.2",
    "node-fetch": "^2.6.0",
    "p-filter": "^2.1.0",
    "p-map": "^4.0.0",
    "p-queue": "^6.4.0",
    "p-retry": "^4.2.0",
    "progress": "^2.0.3",
    "rimraf": "^3.0.2",
    "temp-dir": "^2.0.0",
    "tempy": "^1.0.0"
  },
  "scripts": {
    "build": "tsc -d",
    "prepublishOnly": "pnpm run build",
    "format": "prettier --write .",
    "lint": "eslint --fix --ext .js,.ts .",
    "test": "jest"
  },
  "files": [
    "!**/__tests__",
    "dist"
  ],
  "husky": {
    "hooks": {
      "pre-commit": "lint-staged"
    }
  },
  "lint-staged": {
    "*.{js,ts,css,json,md}": [
      "prettier --write",
      "git add"
    ],
    "*.{js,ts}": [
      "eslint"
    ]
  }
}
