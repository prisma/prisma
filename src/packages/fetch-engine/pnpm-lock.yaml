dependencies:
  '@prisma/debug': 'link:../debug'
  '@prisma/get-platform': 'link:../get-platform'
devDependencies:
  '@prisma/engines-version': 2.11.2-2-7eaf799c2a91e3fb23090404b19cc0398df9bf90
lockfileVersion: 5.2
packages:
  /@prisma/engines-version/2.11.2-2-7eaf799c2a91e3fb23090404b19cc0398df9bf90:
    dev: true
    resolution:
      integrity: sha512-qhD5Mf8B8jc0n+B0pOxvkrq24QPkln94R6Zenf1WAxV/v+X7h9rHjbUSSIBGPULgRqZXJJeP86If3H5FNJ9Ysg==
specifiers:
  '@prisma/debug': 'workspace:*'
  '@prisma/engines-version': 2.11.2-2-7eaf799c2a91e3fb23090404b19cc0398df9bf90
  '@prisma/get-platform': 'workspace:*'
