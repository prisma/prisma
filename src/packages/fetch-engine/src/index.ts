export { download } from './download'
export { getLatestTag, urlExists, getAllUrls } from './getLatestTag'
export { getProxyAgent } from './getProxyAgent'
export { getBinaryName } from './download'
