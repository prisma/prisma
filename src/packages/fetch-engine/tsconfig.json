{
  "compilerOptions": {
    "declaration": true,
    "lib": ["esnext"],
    "module": "commonjs",
    "target": "es2018",
    "strict": true,
    "esModuleInterop": true,
    "sourceMap": true,
    "noImplicitAny": false,
    "outDir": "dist",
    "rootDir": "src",
    "incremental": true
  },
  "exclude": ["dist", "build", "scripts", "sandbox"]
}
