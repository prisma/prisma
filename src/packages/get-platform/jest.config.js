module.exports = {
  preset: 'ts-jest',
  testEnvironment: 'node',
  collectCoverage: true,
  coverageDirectory: 'src/__tests__/coverage',
  collectCoverageFrom: ['src/**/*.ts', '!**/__tests__/**/*'],
  testMatch: ['**/src/__tests__/**/*.test.ts'],
  globals: {
    'ts-jest': {
      packageJson: 'package.json',
    },
  },
}
