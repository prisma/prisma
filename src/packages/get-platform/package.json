{
  "name": "@prisma/get-platform",
  "version": "2.9.0-dev.79",
  "main": "dist/index.js",
  "types": "dist/index.d.ts",
  "license": "Apache-2.0",
  "author": "Tim Suchanek <suchanek@prisma.io>",
  "devDependencies": {
    "@types/jest": "26.0.14",
    "@types/node": "12.12.62",
    "@typescript-eslint/eslint-plugin": "4.2.0",
    "@typescript-eslint/parser": "4.2.0",
    "eslint": "7.10.0",
    "eslint-config-prettier": "6.12.0",
    "eslint-plugin-eslint-comments": "3.2.0",
    "eslint-plugin-jest": "24.0.2",
    "eslint-plugin-prettier": "3.1.4",
    "husky": "4.3.0",
    "jest": "26.4.2",
    "lint-staged": "10.4.0",
    "prettier": "2.1.2",
    "ts-jest": "26.4.0",
    "typescript": "4.0.3"
  },
  "dependencies": {
    "@prisma/debug": "workspace:*"
  },
  "scripts": {
    "build": "tsc -d",
    "prepublishOnly": "pnpm run build && npm run test",
    "format": "prettier --write .",
    "lint": "eslint --fix --ext .js,.ts .",
    "test": "jest"
  },
  "files": [
    "!**/__tests__",
    "dist"
  ],
  "husky": {
    "hooks": {
      "pre-commit": "lint-staged"
    }
  },
  "lint-staged": {
    "*.{js,ts,css,json,md}": [
      "prettier --write",
      "git add"
    ],
    "*.{js,ts}": [
      "eslint"
    ]
  }
}
