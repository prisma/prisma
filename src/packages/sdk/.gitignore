node_modules
query-engine*
src/download-test
test-dmmf.ts
dmmf.json
introspection-engine*
migration-engine*
prisma-fmt*
libquery*
write-test
sandbox
failed-*
/debug.d.ts
/debug.js
/debug.js.map
pnpm-lock.yaml
/engines
