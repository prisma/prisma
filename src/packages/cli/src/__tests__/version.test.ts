import { download } from '@prisma/fetch-engine'
import { getPlatform } from '@prisma/get-platform'
import { engineEnvVarMap } from '@prisma/sdk'
import makeDir from 'make-dir'
import path from 'path'
import { consoleContext, Context } from './__helpers__/context'

const ctx = Context.new().add(consoleContext()).assemble()
const testIf = (condition: boolean) => (condition ? test : test.skip)
const useNodeAPI = process.env.PRISMA_FORCE_NAPI === 'true'
const version = 'e6bd3dc12d849124a04c3a8e6bd9c194381afda3'

describe('version', () => {
  // Node-API Tests

  testIf(useNodeAPI)('basic version (Node-API)', async () => {
    const data = await ctx.cli('--version')
    expect(cleanSnapshot(data.stdout)).toMatchSnapshot()
  })

  testIf(useNodeAPI)(
    'version with custom binaries (Node-API)',
    async () => {
      const enginesDir = path.join(__dirname, 'version-test-engines')
      await makeDir(enginesDir)
      const binaryPaths = await download({
        binaries: {
          'introspection-engine': enginesDir,
          'migration-engine': enginesDir,
          'prisma-fmt': enginesDir,
          'libquery-engine': enginesDir,
        },
        version,
        failSilent: false,
      })
      // This Omits query-engine from the map
      const { ['query-engine']: qe, ...envVarMap } = engineEnvVarMap

      const platform = await getPlatform()

      for (const engine in envVarMap) {
        const envVar = envVarMap[engine]
        process.env[envVar] = binaryPaths[engine][platform]
        // console.debug(`Setting ${envVar} to ${binaryPaths[engine][platform]}`)
      }

      const data = await ctx.cli('--version')
      expect(cleanSnapshot(data.stdout)).toMatchSnapshot()

      // cleanup
      for (const engine in envVarMap) {
        const envVar = envVarMap[engine]
        delete process[envVar]
      }
    },
    20000,
  )

  // Binary Tests

  testIf(!useNodeAPI)('basic version', async () => {
    const data = await ctx.cli('--version')
    expect(cleanSnapshot(data.stdout)).toMatchSnapshot()
  })

  testIf(!useNodeAPI)(
    'version with custom binaries',
    async () => {
      const enginesDir = path.join(__dirname, 'version-test-engines')
      await makeDir(enginesDir)
      const binaryPaths = await download({
        binaries: {
          'introspection-engine': enginesDir,
          'migration-engine': enginesDir,
          'prisma-fmt': enginesDir,
          'query-engine': enginesDir,
        },
        version,
        failSilent: false,
      })

      const platform = await getPlatform()
      const { ['libquery-engine']: qe, ...envVarMap } = engineEnvVarMap
      for (const engine in envVarMap) {
        const envVar = envVarMap[engine]
        process.env[envVar] = binaryPaths[engine][platform]
        // console.debug(`Setting ${envVar} to ${binaryPaths[engine][platform]}`)
      }

      const data = await ctx.cli('--version')
      expect(cleanSnapshot(data.stdout)).toMatchSnapshot()

      // cleanup
      for (const engine in envVarMap) {
        const envVar = envVarMap[engine]
        delete process[envVar]
      }
    },
    20000,
  )
})

function cleanSnapshot(str: string): string {
  return str.replace(/:(.*)/g, ': placeholder')
}
