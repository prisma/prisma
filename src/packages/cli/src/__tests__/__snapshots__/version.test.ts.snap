// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`version basic version (Node-API) 1`] = `
prisma                  : placeholder
@prisma/client          : placeholder
Current platform        : placeholder
Query Engine (Node-API) : placeholder
Migration Engine        : placeholder
Introspection Engine    : placeholder
Format Binary           : placeholder
Default Engines Hash    : placeholder
Studio                  : placeholder
`;

exports[`version basic version 1`] = `
prisma               : placeholder
@prisma/client       : placeholder
Current platform     : placeholder
Query Engine         : placeholder
Migration Engine     : placeholder
Introspection Engine : placeholder
Format Binary        : placeholder
Default Engines Hash : placeholder
Studio               : placeholder
`;

exports[`version version with custom binaries (Node-API) 1`] = `
prisma                  : placeholder
@prisma/client          : placeholder
Current platform        : placeholder
Query Engine (Node-API) : placeholder
Migration Engine        : placeholder
Introspection Engine    : placeholder
Format Binary           : placeholder
Default Engines Hash    : placeholder
Studio                  : placeholder
`;

exports[`version version with custom binaries 1`] = `
prisma               : placeholder
@prisma/client       : placeholder
Current platform     : placeholder
Query Engine         : placeholder
Migration Engine     : placeholder
Introspection Engine : placeholder
Format Binary        : placeholder
Default Engines Hash : placeholder
Studio               : placeholder
`;
